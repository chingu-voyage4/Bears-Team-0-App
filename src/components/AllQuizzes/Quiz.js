import React, { Component } from "react";
class Quiz extends Component {
  render() {
    return (
      <div className="quiz">
        <h1>{this.props.title}</h1>
        <p>{this.props.body}</p>
        <p>
          <span>
            <i className="fa fa-heart" /> {this.props.likes}
          </span>{" "}
          <a href="#">Share</a>{" "}
        </p>
      </div>
    );
  }
}

export default Quiz;
