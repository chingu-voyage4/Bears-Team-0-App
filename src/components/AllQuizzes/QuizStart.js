import React, { Component } from "react";

class QuizStart extends Component {
  render() {
    return <div />;
  }
}

export default QuizStart;
