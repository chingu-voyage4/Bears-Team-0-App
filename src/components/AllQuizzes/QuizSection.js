import React, { Component } from "react";
import Quiz from "./Quiz";

export class QuizSection extends Component {
  componentWillMount() {
    this.props.getData();
  }
  render() {
    return (
      <div className="quiz-section">
        <h1>{this.props.headingText}</h1>
        {this.props.quizzes
          ? this.props.quizzes.map(e => (
              <Quiz key={e.id} title={e.title} likes={e.likes} body={e.body} />
            ))
          : null}
      </div>
    );
  }
}

export default QuizSection;
