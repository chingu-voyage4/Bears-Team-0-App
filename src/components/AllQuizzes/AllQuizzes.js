import React, { Component } from "react";
import PropTypes from "prop-types";
import QuizSection from "./QuizSection";
import QuizStart from "./QuizStart";
import { fetchAllQuizzes, fetchYourQuizzes } from "../../actions/quizzes";
import { connect } from "react-redux";

export class AllQuizzes extends Component {
  render() {
    return (
      <div className="all-quizzes">
        <section className="all-quizzes-left">
          <QuizStart />
        </section>
        <section className="all-quizzes-right">
          <QuizSection
            headingText="Your Quizzes"
            quizzes={this.props.yourQuizzes}
            getData={this.props.fetchYourQuizzes}
            wrap={false}
            mainColor="secondary"
          />
          <QuizSection
            headingText="Popular Quizzes"
            quizzes={this.props.popularQuizzes}
            getData={this.props.fetchAllQuizzes}
            wrap={true}
            mainColor="tertiary"
          />
        </section>
      </div>
    );
  }
}

export default connect(
  state => {
    return {
      popularQuizzes: state.allQuizzes.popularQuizzes,
      yourQuizzes: state.allQuizzes.yourQuizzes
    };
  },
  {
    fetchAllQuizzes: fetchAllQuizzes,
    fetchYourQuizzes: fetchYourQuizzes
  }
)(AllQuizzes);
