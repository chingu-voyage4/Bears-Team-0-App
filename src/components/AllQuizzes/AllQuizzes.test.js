import React from "react";
import ReactDOM from "react-dom";
import { AllQuizzes } from "./AllQuizzes";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

describe("AllQuizzes", () => {
  it("renders without crashing", () => {
    const wrapper = shallow(<AllQuizzes />);
    expect(wrapper).toMatchSnapshot();
  });
});
