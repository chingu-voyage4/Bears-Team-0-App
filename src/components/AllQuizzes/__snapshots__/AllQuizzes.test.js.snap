// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllQuizzes renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AllQuizzes />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <section
          className="all-quizzes-left"
>
          <Connect(QuizStart) />
</section>,
        <section
          className="all-quizzes-right"
>
          <QuizSection
                    getData={undefined}
                    headingText="Your Quizzes"
                    mainColor="secondary"
                    quizzes={undefined}
                    wrap={false}
          />
          <QuizSection
                    getData={undefined}
                    headingText="Popular Quizzes"
                    mainColor="tertiary"
                    quizzes={undefined}
                    wrap={true}
          />
</section>,
      ],
      "className": "all-quizzes",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Connect(QuizStart) />,
          "className": "all-quizzes-left",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <QuizSection
              getData={undefined}
              headingText="Your Quizzes"
              mainColor="secondary"
              quizzes={undefined}
              wrap={false}
/>,
            <QuizSection
              getData={undefined}
              headingText="Popular Quizzes"
              mainColor="tertiary"
              quizzes={undefined}
              wrap={true}
/>,
          ],
          "className": "all-quizzes-right",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "getData": undefined,
              "headingText": "Your Quizzes",
              "mainColor": "secondary",
              "quizzes": undefined,
              "wrap": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "getData": undefined,
              "headingText": "Popular Quizzes",
              "mainColor": "tertiary",
              "quizzes": undefined,
              "wrap": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "section",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <section
            className="all-quizzes-left"
>
            <Connect(QuizStart) />
</section>,
          <section
            className="all-quizzes-right"
>
            <QuizSection
                        getData={undefined}
                        headingText="Your Quizzes"
                        mainColor="secondary"
                        quizzes={undefined}
                        wrap={false}
            />
            <QuizSection
                        getData={undefined}
                        headingText="Popular Quizzes"
                        mainColor="tertiary"
                        quizzes={undefined}
                        wrap={true}
            />
</section>,
        ],
        "className": "all-quizzes",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(QuizStart) />,
            "className": "all-quizzes-left",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <QuizSection
                getData={undefined}
                headingText="Your Quizzes"
                mainColor="secondary"
                quizzes={undefined}
                wrap={false}
/>,
              <QuizSection
                getData={undefined}
                headingText="Popular Quizzes"
                mainColor="tertiary"
                quizzes={undefined}
                wrap={true}
/>,
            ],
            "className": "all-quizzes-right",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "getData": undefined,
                "headingText": "Your Quizzes",
                "mainColor": "secondary",
                "quizzes": undefined,
                "wrap": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "getData": undefined,
                "headingText": "Popular Quizzes",
                "mainColor": "tertiary",
                "quizzes": undefined,
                "wrap": true,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "section",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
